source/exec.tex

```diff
@@ -0,0 +1,5466 @@
+
+\rSec2[exec.queryable.general]{General}
+
+\pnum
+A \defnadj{queryable}{object} is
+a read-only collection of key/value pair
+where each key is a customization point object known as a \defn{query object}.
+A \defn{query} is an invocation of a query object
+with a queryable object as its first argument and
+a (possibly empty) set of additional arguments.
+A query imposes syntactic and semantic requirements on its invocations.
+
+\pnum
+Let \tcode{q} be a query object,
+let \tcode{args} be a (possibly empty) pack of subexpressions,
+let \tcode{env} be a subexpression
+that refers to a queryable object \tcode{o} of type \tcode{O}, and
+let \tcode{cenv} be a subexpression referring to \tcode{o}
+such that \tcode{decltype((cenv))} is \tcode{const O\&}.
+The expression \tcode{q(env, args...)} is equal to\iref{concepts.equality}
+the expression \tcode{q(cenv, args...)}.
+
+\pnum
+The type of a query expression cannot be \tcode{void}.
+
+\pnum
+The expression \tcode{q(env, args...)} is
+equality-preserving\iref{concepts.equality} and
+does not modify the query object or the arguments.
+
+\pnum
+If the expression \tcode{env.query(q, args...)} is well-formed,
+then it is expression-equivalent to \tcode{q(env, args...)}.
+
+\pnum
+Unless otherwise specified,
+the result of a query is valid as long as the queryable object is valid.
+
+\rSec2[exec.queryable.concept]{\tcode{queryable} concept}
+
+\begin{codeblock}
+namespace std {
+  template<class T>
+    concept @\defexposconcept{queryable}@ = @\libconcept{destructible}@<T>;   // \expos
+}
+\end{codeblock}
+
+\pnum
+The exposition-only \exposconcept{queryable} concept specifies
+the constraints on the types of queryable objects.
+
+\pnum
+Let \tcode{env} be an object of type \tcode{Env}.
+The type \tcode{Env} models queryable
+if for each callable object \tcode{q} and a pack of subexpressions \tcode{args},
+if \tcode{requires \{ q(env, args...) \}} is \tcode{true} then
+\tcode{q(env, args...)} meets any semantic requirements imposed by \tcode{q}.
+
+\rSec1[exec.async.ops]{Asynchronous operations}
+
+\pnum
+An \defnadj{execution}{resource} is a program entity that manages
+a (possibly dynamic) set of execution agents\iref{thread.req.lockable.general},
+which it uses to execute parallel work on behalf of callers.
+\begin{example}
+The currently active thread,
+a system-provided thread pool, and
+uses of an API associated with an external hardware accelerator
+are all examples of execution resources.
+\end{example}
+Execution resources execute asynchronous operations.
+An execution resource is either valid or invalid.
+
+\pnum
+An \defnadj{asynchronous}{operation} is
+a distinct unit of program execution that
+\begin{itemize}
+\item
+is explicitly created;
+\item
+can be explicitly started once at most;
+\item
+once started, eventually completes exactly once
+with a (possibly empty) set of result datums and
+in exactly one of three \defnx{disposition}{dispositions}:
+success, failure, or cancellation;
```

Comment:
in the pdf, this is rendered as "_disposition_" (singlular) should be plural.
![image](https://github.com/user-attachments/assets/fa9cee88-3fd4-42c6-94e2-6467128fe2cd)

Reply:
Fixing, thanks!